utils/s3/client.cc

```diff
@@ -385,10 +403,110 @@ class client::upload_sink final : public client::upload_sink_base {
     }
 };
 
+future<> client::upload_sink_base::upload_part(std::unique_ptr<upload_sink> piece_ptr) {
+    if (!upload_started()) {
+        co_await start_upload();
+    }
+
+    auto& piece = *piece_ptr;
+    unsigned part_number = _part_etags.size();
+    _part_etags.emplace_back();
+    s3l.trace("PUT part {} from {} (upload id {})", part_number, piece._object_name, _upload_id);
+    auto req = http::request::make("PUT", _client->_host, _object_name);
+    req.query_parameters["partNumber"] = format("{}", part_number);
```

Comment:
> part_number + 1?

I have spawned this branch _before_ #13534 got merged, so here -- no, not +1, otherwise `dump_multipart_upload_parts` won't work, they are in sync.